### PR TITLE
1436790/1438462: Part 14: 20 files with MSDN link fixes.

### PR DESCRIPTION
**Pull request by DennisLee-DennisLee**
Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037) and applies to the following work items:
- [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
- [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

### CODE REVIEW

File: docs/framework/data/adonet/queries-in-linq-to-dataset.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ A query is an expression that retrieves data from a data source. Queries are usu
...
   
 ### Query Expression Syntax  
- Query expressions are a declarative query syntax. This syntax enables a developer to write queries in C# or Visual Basic in a format similar to SQL. By using query expression syntax, you can perform even complex filtering, ordering, and grouping operations on data sources with minimal code. For more information, see [LINQ Query Expressions](https://msdn.microsoft.com/library/40638f19-fb46-4d26-a2d9-a383b48f5ed4) and [Basic Query Operations (Visual Basic)](~/docs/visual-basic/programming-guide/concepts/linq/basic-query-operations.md).  
+ Query expressions are a declarative query syntax. This syntax enables a developer to write queries in C# or Visual Basic in a format similar to SQL. By using query expression syntax, you can perform even complex filtering, ordering, and grouping operations on data sources with minimal code. For more information, see [LINQ Query Expressions](../../../csharp/linq/index.md#query-expression-overview) and [Basic Query Operations (Visual Basic)](~/docs/visual-basic/programming-guide/concepts/linq/basic-query-operations.md).
```
Comment (mairaw) [inline docs/framework/data/adonet/queries-in-linq-to-dataset.md]:
```suggestion
 Query expressions are a declarative query syntax. This syntax enables a developer to write queries in C# or Visual Basic in a format similar to SQL. By using query expression syntax, you can perform even complex filtering, ordering, and grouping operations on data sources with minimal code. For more information, see [LINQ Query Expressions](../../../csharp/linq/index.md#query-expression-overview) and [Basic Query Operations (Visual Basic)](../../../visual-basic/programming-guide/concepts/linq/basic-query-operations.md).
```

File: docs/framework/data/adonet/query-expression-examples-linq-to-dataset.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ ms.date: "03/30/2017"
...
+This section provides [!INCLUDE[linq_dataset](../../../../includes/linq-dataset-md.md)] programming examples in query expression syntax that use the standard query operators. The <xref:System.Data.DataSet> used in these examples is populated by using the `FillDataSet` method, which is specified in [Loading Data Into a DataSet](../../../../docs/framework/data/adonet/loading-data-into-a-dataset.md). For more information, see [Standard Query Operators Overview (C#)](../../../csharp/programming-guide/concepts/linq/standard-query-operators-overview.md) or [Standard Query Operators Overview (Visual Basic)](../../../visual-basic/programming-guide/concepts/linq/standard-query-operators-overview.md).
```
Comment (mairaw) [inline docs/framework/data/adonet/query-expression-examples-linq-to-dataset.md]:
```suggestion
This section provides LINQ to DataSet programming examples in query expression syntax that use the standard query operators. The <xref:System.Data.DataSet> used in these examples is populated by using the `FillDataSet` method, which is specified in [Loading Data Into a DataSet](../../../../docs/framework/data/adonet/loading-data-into-a-dataset.md). For more information, see [Standard Query Operators Overview (C#)](../../../csharp/programming-guide/concepts/linq/standard-query-operators-overview.md) or [Standard Query Operators Overview (Visual Basic)](../../../visual-basic/programming-guide/concepts/linq/standard-query-operators-overview.md).  
```

File: docs/framework/data/adonet/secure-client-applications.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ Applications typically consist of many parts that must all be protected from vul
  [Windows Forms and Unmanaged Applications](../../../../docs/framework/winforms/advanced/windows-forms-and-unmanaged-applications.md)  
  Describes how to interact with unmanaged applications in a Windows Forms application.  
   
- [ClickOnce Deployment for Windows Forms Applications](https://msdn.microsoft.com/library/34d8c770-48f2-460c-8d67-4ea5684511df)  
+ [ClickOnce Deployment for Windows Forms Applications](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2008/wh45kb66(v=vs.90))
```
Comment (mairaw) [inline docs/framework/data/adonet/secure-client-applications.md]:
```suggestion
 [ClickOnce Deployment for Windows Forms](../../winforms/clickonce-deployment-for-windows-forms.md)  
```

Comment (mairaw) [reply docs/framework/data/adonet/secure-client-applications.md]: This same link is also used in another topic: make the same fix there (it doesn't need to be in this same PR): https://github.com/dotnet/docs/blob/master/docs/framework/data/adonet/sql/sql-server-express-user-instances.md

File: docs/framework/data/adonet/sql/clr-triggers.md [hunk 4]
```diff
@@ -13,5 +13,5 @@ A trigger is a special type of stored procedure that automatically runs when a l
 1. [CLR Triggers](/sql/database-engine/dev-guide/clr-triggers)
   
 ## See also
-- [Creating SQL Server 2005 Objects In Managed Code](https://msdn.microsoft.com/library/5358a825-e19b-49aa-8214-674ce5fed1da)
+- [Creating SQL Server 2005 Objects In Managed Code](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2008/6s0s2at1(v=vs.90))
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/clr-triggers.md]: let's remove this one 

```suggestion

```

File: docs/framework/data/adonet/sql/enabling-cross-database-access-in-sql-server.md [hunk 5]
```diff
@@ -42,7 +42,7 @@ ALTER DATABASE Database2 SET DB_CHAINING ON;
   
 |Resource|Description|  
 |--------------|-----------------|  
-|[Extending Database Impersonation by Using EXECUTE AS](https://msdn.microsoft.com/library/ms188304\(SQL.105\).aspx) and [Cross DB Ownership Chaining Option](/sql/database-engine/configure-windows/cross-db-ownership-chaining-server-configuration-option)SQL Server Books Online.|Topics describe how to configure cross-database ownership chaining for an instance of SQL Server.|  
+|[Extending Database Impersonation by Using EXECUTE AS](https://docs.microsoft.com/previous-versions/sql/sql-server-2008-r2/ms188304(v=sql.105)) and [Cross DB Ownership Chaining Option](/sql/database-engine/configure-windows/cross-db-ownership-chaining-server-configuration-option)SQL Server Books Online.|Topics describe how to configure cross-database ownership chaining for an instance of SQL Server.|
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/enabling-cross-database-access-in-sql-server.md]:
```suggestion
|[Extending Database Impersonation by Using EXECUTE AS](https://docs.microsoft.com/previous-versions/sql/sql-server-2008-r2/ms188304(v=sql.105)) and [Cross DB Ownership Chaining Option](/sql/database-engine/configure-windows/cross-db-ownership-chaining-server-configuration-option).|Articles describe how to configure cross-database ownership chaining for an instance of SQL Server.|  
```

**Review by mairaw — CHANGES_REQUESTED** [review]
Looks good @DennisLee-DennisLee. A few suggestions before we merge.

**Comment by DennisLee-DennisLee** [comment]
#sign-off

**Review by mairaw — APPROVED** [review]
LGTM. Thanks @DennisLee-DennisLee!

**Pull request merged**